File: Cpp_chunks/sudoku_grid.cpp
class SudokuGrid {
private:
    int n ;
    vector<int> I, Omega;
    vector<vector<SudokuCell>> grid;
    int boxOf(int row, int col){
        int root = sqrt(n);
        return root*(row/root)+col/root;
    }
    void resizeGrid(){
        vector<SudokuCell> rowVector;
        rowVector.resize(n, SudokuCell());
        grid.resize(n, rowVector);
    }
    void initializeGrid(vector<vector<int>> &numberGrid){
        int value;
        for(int row: I){
            for(int col: I){
                value = numberGrid[row][col];
                if(value != 0) {
                    grid[row][col].value = value;
                    grid[row][col].fixed = true;
                }
                else{
                    grid[row][col].avSet = new Set(n);
                }
            }
        }
    }
public:
    SudokuGrid(vector<vector<int>> numberGrid, int n){
        SudokuGrid::n = n;
        I = vector<int>(n);   iota(I.begin(), I.end(), 0);
        Omega = vector<int>(n);   iota(Omega.begin(), Omega.end(), 1);
        images = new BracketImage(n);
        resizeGrid();
        initializeGrid(numberGrid);
    }
};

File: Python/chunks/sudoku_grid.py
class SudokuGrid:
    n = 9
    I = [i for i in range(n)]
    Omega = [i for i in range(1,n+1)]

    def box_of(self, row, col):
        return 3 * int(row / 3) + int(col / 3)

    def __init__(self, number_grid):
        self.grid = [[SudokuCell(num, True) if num != 0 else SudokuCell() for num in row] for row in number_grid]
        for row in self.I:
            for col in self.I:
                cell = self.grid[row][col]
                if not cell.has_value():
                    for val in self.Omega:
                        cell.av_set.add(val)